pkg/controller/machineset/controller.go

```diff
@@ -342,3 +332,41 @@ func getMachinesToDelete(filteredMachines []*v1alpha1.Machine, diff int) []*v1al
 	// see: https://github.com/kubernetes/kube-deploy/issues/625
 	return filteredMachines[:diff]
 }
+
+func (c *MachineSetControllerImpl) waitForMachineCreation(machineList []*v1alpha1.Machine) error {
+	for _, machine := range machineList {
+		pollErr := util.Poll(stateConfirmationInterval, stateConfirmationTimeout, func() (bool, error) {
+			_, err := c.machineLister.Machines(machine.Namespace).Get(machine.Name)
+			glog.Error(err)
+			if err == nil {
+				return true, nil
+			}
+			if errors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
```

Comment:
Please log all errors explicitly for debugging

Reply:
error from the retry loop goes to pollErr which is always logged.